app/src/main/java/org/onionshare/android/tor/TorManager.kt

```diff
@@ -64,118 +60,92 @@ class TorManager @Inject constructor(
      * Starts [ShareService] and creates a new onion service.
      * Suspends until the address of the onion service is available.
      */
+    @Throws(IOException::class, IllegalArgumentException::class, InterruptedException::class)
     suspend fun start() = withContext(Dispatchers.IO) {
-        if (state.value !is TorState.Stopped) stop()
-
         LOG.info("Starting...")
         val now = System.currentTimeMillis()
         _state.value = TorState.Starting(progress = 0, startTime = now, lastProgressTime = now)
-
         Intent(app, ShareService::class.java).also { intent ->
             startForegroundService(app, intent)
         }
 
         tor.start()
+        changeStartingState(5)
+        if (settingsManager.automaticBridges.value) {
+            startCheckJob = launch {
+                LOG.info("Starting check job")
+                checkStartupProgress()
+                LOG.info("Check job finished")
+            }
+        } else {
+            val customBridges = settingsManager.customBridges.value
+            if (customBridges.isNotEmpty()) {
+                LOG.info("Using ${customBridges.size} custom bridges...")
+                tor.enableBridges(customBridges.map { "Bridge $it" })
+            }
+        }
+        tor.enableNetwork(true)
     }
 
-    fun stop(failedToConnect: Boolean = false) {
+    fun stop() {
         LOG.info("Stopping...")
         startCheckJob?.cancel()
         startCheckJob = null
-        _state.value = TorState.Stopping(failedToConnect)
         tor.stop()
         Intent(app, ShareService::class.java).also { intent ->
             app.stopService(intent)
         }
     }
 
-    @OptIn(DelicateCoroutinesApi::class)
     override fun onState(s: TorWrapper.TorState) {
```

Comment:
Could we save a bit of boilerplate by doing `LOG.info("new state: $s")` outside the `when`?

Reply:
Yes, good point. We did more here before, but if we won't be doing anything else, we might as well remove the when branches.